src/components/app.js

```diff
@@ -111,8 +111,8 @@ const App = () => {
             });
     }, []);
 
-    function studyClickHandler(studyName) {
-        history.push('/studies/' + encodeURIComponent(studyName));
+    function studyClickHandler(studyName, userId) {
+        history.push('/studies/' + encodeURIComponent(studyName) + '/' + encodeURIComponent(userId.subject) + '/' + encodeURIComponent(userId.issuer));
```

Comment:
Not sure what to do here:
- keep a short form url with only the studyname in urls and assume the currently logged in user/issuer; in addition to having a longer form with the user/issuer that's shareable with other users
- always use the longer form

Anyway I think the url should be /studies/issuer/user/studyname

Reply:
This seems to me an incorrect REST api design: studies/{studyName} is unseccable. 
I would use query parameters as there is no hierarchy beetwen issuer, user and studies
/studies/{studyName}?issuer=,user=

Reply:
I don't feel like it's a good idea either to use query parameters as the base url should refer to a specific resource and it's not the case if we do : studies/{studyName}

Reply:
Why? 
studies/studyName is a resource

Reply:
but you cannot make a link between a studyName and a study anymore. The issuer and subject are mandatory to get a study and therefore should be a path variable IMO.